src/actors/objects/mod.rs

```diff
@@ -552,29 +596,44 @@ impl ObjectsActor {
         file_metas.and_then(move |responses: Vec<Result<Arc<ObjectFileMeta>, _>>| {
             responses
                 .into_iter()
-                .enumerate()
-                .min_by_key(|(i, response)| {
+                .min_by_key(|response| {
                     // Prefer files that contain an object over unparseable files
-                    let object = match response {
-                        Ok(object) => object,
+                    let object_meta = match response {
+                        Ok(object_meta) => object_meta,
                         Err(e) => {
                             log::debug!("Failed to download: {:#?}", e);
-                            return (3, *i);
+                            return 3;
                         }
                     };
 
                     // Prefer object files with debug/unwind info over object files without
-                    let score = match purpose {
-                        ObjectPurpose::Unwind if object.features.has_unwind_info => 0,
-                        ObjectPurpose::Debug if object.features.has_debug_info => 0,
-                        ObjectPurpose::Debug if object.features.has_symbols => 1,
-                        ObjectPurpose::Source if object.features.has_sources => 0,
+                    match purpose {
+                        ObjectPurpose::Unwind if object_meta.features.has_unwind_info => 0,
+                        ObjectPurpose::Debug if object_meta.features.has_debug_info => 0,
+                        ObjectPurpose::Debug if object_meta.features.has_symbols => 1,
+                        ObjectPurpose::Source if object_meta.features.has_sources => 0,
                         _ => 2,
-                    };
-
-                    (score, *i)
+                    }
+                })
+                .filter(|response| match response {
+                    // Make sure we only return objects which provide the requested info
```

Comment:
Can we dedupe this with the content in `min_by_key`? iiuc all scores of 2 should basically be filtered out.

Perhaps the closure of `min_by_key` should go into a `filter_map` before `min_by` happens. Or we refactor this entire thing into a regular for-loop.

Reply:
Thanks for complaining about this.  I've looked at all the consumers again and concluded the filtering needs to be moved to before the `min_by_key` in order to still return a possible `Err` if we filtered out an object file.

I'm not sure I would like to merge the two, I think having the two as separate operations is easier to understand as each have a defined scope.

Reply:
Yeah no worries was just an idea.